arm/Microsoft.AppConfiguration/configurationStores/deploy.bicep

```diff
@@ -188,13 +188,23 @@ module appConfiguration_rbac '.bicep/nested_rbac.bicep' = [for (roleAssignment,
   }
 }]
 
-module appConfiguration_privateEndpoints '.bicep/nested_privateEndpoint.bicep' = [for (privateEndpoint, index) in privateEndpoints: {
-  name: '${uniqueString(deployment().name, location)}-AppConfig-PrivateEndpoint-${index}'
+module appConfiguration_privateEndpoints '../../Microsoft.Network/privateEndpoints/deploy.bicep' = [for (privateEndpoint, index) in privateEndpoints: {
+  name: '${uniqueString(deployment().name, location)}-appConfiguration-PrivateEndpoint-${index}'
   params: {
-    privateEndpointResourceId: appConfiguration.id
-    privateEndpointVnetLocation: empty(privateEndpoints) ? 'dummy' : reference(split(privateEndpoint.subnetResourceId, '/subnets/')[0], '2020-06-01', 'Full').location
-    privateEndpointObj: privateEndpoint
-    tags: tags
+    groupIds: [
+      privateEndpoint.service
+    ]
+    name: contains(privateEndpoint, 'name') ? privateEndpoint.name : '${privateEndpoint.service}-${last(split(appConfiguration.id, '/'))}-${split(privateEndpoint.subnetResourceId, '/')[8]}_${last(split(privateEndpoint.subnetResourceId, '/'))}'
```

Comment:
I built this after a patter we used at a customer. The idea was to have some sort of idea what kind of endpoint it is. Hence the service name, target subnet name & groupId. I'm happy to do this differently, but I wonder if we should discuss it in this PR or a separate issue. If in this PR, I'd ask e.g. @eriqua & @MariusStorhaug to also give it some thought. :)

Reply:
@eriqua @MrMCake just looked at the names we had before:
`'${privateEndpointResourceName}-${privateEndpointObj.service}'` this is from the nested_privateEndpoint.bicep.

and on the main `privateEndpoints` module, we have been enforcing the name as a 'required' parameter

So I think if we are moving to a new default, we should have some alignment on it, unless we want to make it required?

Reply:
Correct. However, whatever we do, we have to ensure not 2 resources endpoints can have the same name.

Reply:
Would 

pe-${last(split(<<resourceId>>, '/'))}-${privateEndpoint.service}-${index+1}

Or 

${last(split(<<resourceId>>, '/'))}-${privateEndpoint.service}-pe-${index+1}

Be an option on the modules that leverage the PE module.. and we keep the default on the main PE module to "require" a name?

Reply:
Would

```bicep
pe-${last(split(<<resourceId>>, '/'))}-${privateEndpoint.service}-${index+1}

```
or

```bicep
${last(split(<<resourceId>>, '/'))}-${privateEndpoint.service}-pe-${index+1}
```
Be an option on the modules that leverage the PE module.. and we keep the default on the main PE module to "require" a name?

Reply:
In case we end up having a required name on the pe module and that name being passed from modules cross referencing that, what about a `Conditional` category, similar to what we have for child modules' parent, `Required if the template is used in a standalone deployment.` ?

Reply:
> In case we end up having a required name on the pe module and that name being passed from modules cross referencing that, what about a `Conditional` category, similar to what we have for child modules' parent, `Required if the template is used in a standalone deployment.` ?

As the invoking module has its own parameter usage block for the private endpoint I think we can avoid that. Maybe im interpreting to much into it, but in general I'd like to try and keep this to a mimum anyways as all our modules could in theory be invoked by one another - so the 'condition' could be added to all modules.

Reply:
@eriqua I removed the default name now from the PE module
@ahmadabdalla Updated the naming in the modules to your suggestion

Reply:
Thanks @MrMCake .. noting its using `index` not `index+1`, but I'm ok with that as well. @eriqua please feel free to resolve if you don't have further comments